modules/covid_vaccine/app/services/covid_vaccine/v0/expanded_registration_service.rb

```diff
@@ -0,0 +1,144 @@
+        # Conditions where we should not send data to vetext:
+        # 1 - no preferred location in raw_form_data and no facility found for zip_code: manual intervention
+        # 2 - no ICN found in MPI: retry
+        # 3 - Station ID returned from MPI is different than preferred location: retry
+        
+        vetext_attributes = {}
+        
+        # preferred facility will either be in eligibility_info, or raw_form_data. If its in neither one,
+        # for the purposes of this register method we should not be fetching facilities and trying to reconcile;
+        # instead we will set the state to :enrollment_out_of_band
+        
+        # Get the preferred facility here as it is needed in MPI lookup
+        # facility = submission&.eligibility_info&['preferred_facility'] || raw_form_data['preferred_facility'].delete_prefix('vha_')
+        facility = submission&.eligibility_info&.dig('preferred_facility') || raw_form_data['preferred_facility'].delete_prefix('vha_')
+
+
+        if facility.blank?
+          submission.enrollment_requires_intervention!
+          return
+        end
+        sta3n = facility[0..2]
+        sta6a = facility if facility.length > 3
+        
+        vetext_attributes.merge!(form_attributes(raw_form_data, sta3n, sta6a))
+        
+        # MPI Query must succeed and return ICN and expected facilityID before we send this data to backend service
+        mpi_attributes = attributes_from_mpi(raw_form_data, sta3n)
+        
+        if mpi_attributes[:error].present?
+          Rails.logger.info(
+            "#{self.class.name}:Error in MPI Lookup",
+            mpi_error: mpi_attributes[:error],
+            submission: submission.id
+          )
+          return
+        else
+          submission.detected_enrollment!
+          vetext_attributes.merge!(mpi_attributes).compact!
+        end
+        submit_and_save(vetext_attributes, submission, user_type)
+      end
+      
+      private
+      
+      def submit_and_save(attributes, submission, user_type)
+        # TODO: error handling
+        audit_log(attributes, user_type)
+        response = submit(attributes)
+        Rails.logger.info("Covid_Vaccine_Expanded Vetext Response: #{response}")
+        elig_info_icn = {'patient_icn': attributes[:patient_icn]}
+        elig_info_icn.merge!(submission.eligibility_info) unless submission.eligibility_info == nil
+       
+        submission.update!(vetext_sid: response[:sid], form_data: attributes, state: 'registered', eligibility_info: elig_info_icn)
+        submit_confirmation_email(attributes[:email], submission.created_at, response[:sid])
+        submission
+      end
+      
+      def submit_confirmation_email(email, date, sid)
+        return if email.blank?
+        
+        formatted_date = date.strftime('%B %-d, %Y %-l:%M %P %Z').sub(/([ap])m/, '\1.m.')
+        CovidVaccine::ExpandedRegistrationEmailJob.perform_async(email, formatted_date, sid)
```

Comment:
Is this the right mailer? I think this is the one we were using when the form is received.

Reply:
maybe this code was already here for submission received, but i want to emphasize we're not notifying veterans beyond submission received which i think was handled in earlier PRs. We do not need to notify a veteran that they have "registered."